File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int depth;
    void dfs(TreeNode* root, int curDepth){
        if(root == NULL){
            depth = max(depth, curDepth);
            return;
        }
        curDepth++;
        //depth = max(depth, curDepth);
        dfs(root->left, curDepth);
        dfs(root->right,curDepth);
    }
    int maxDepth(TreeNode* root) {
        int curDepth;
        depth = curDepth = 0;
        dfs(root, curDepth);
        return depth;
    }
};

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def dfs(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0
        return 1 + max(self.dfs(root.left), self.dfs(root.right))
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        return self.dfs(root)
        